lib/shape.ml

```diff
@@ -13,6 +14,10 @@ let axes_flag = ref false
 let draw_axes flag = 
   axes_flag := flag
 
+let draw_line line = 
+  moveto line.a.x line.a.y;
+  lineto line.b.x line.b.y
```

Comment:
I am aware that the library has lots of stateful elements, but I feel we should stay from them as much as possible.

There is an alternative to `moveto` and `lineto`, which is `draw_poly_line`. We should use that instead.

Reply:
100% in agreement wrt state, wasn't sure how y'all wanted to do things, I'll refactor this tonight :)